pkg/apis/serving/k8s_validation.go

```diff
@@ -425,11 +462,34 @@ func validateProbe(p *corev1.Probe) *apis.FieldError {
 	h := p.Handler
 	errs = errs.Also(apis.CheckDisallowedFields(h, *HandlerMask(&h)))
 
-	switch {
-	case h.HTTPGet != nil:
-		errs = errs.Also(apis.CheckDisallowedFields(*h.HTTPGet, *HTTPGetActionMask(h.HTTPGet))).ViaField("httpGet")
-	case h.TCPSocket != nil:
-		errs = errs.Also(apis.CheckDisallowedFields(*h.TCPSocket, *TCPSocketActionMask(h.TCPSocket))).ViaField("tcpSocket")
+	numHandlers := 0
+
+	if h.HTTPGet != nil {
+		if numHandlers > 0 {
+			errs = errs.Also(apis.ErrDisallowedFields("httpGet"))
```

Comment:
This code is unreachable since `numHandlers := 0` on  line 465

Reply:
Yeah, the unreachable code is responsible for the above drop in code coverage. The implementation is lifted from how k8s validates its probes. I figured the check was there as a protection against some day in the far off future where we carelessly added another handler, but that may be a little much. 

Happy to pull it out if we don't want to worry about this.

Reply:
Ah I see. Yeah I think it's more straightforward if we remove this.